pkgs/by-name/ot/otadump/package.nix

```diff
@@ -0,0 +1,40 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  protobuf,
+  xz,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "otadump";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "crazystylus";
+    repo = pname;
+    rev = version;
+    hash = "sha256-4zPVcTU+0otV4EPQi80uSRkpRo9XzI0V3Kr17ugXX2U=";
+  };
+
+  patches = [ ./no-static.patch ];
+
+  nativeBuildInputs = [
+    pkg-config
+    protobuf
+  ];
+
+  buildInputs = [ xz.dev ];
+
+  doCheck = false; # There are no tests
+
+  cargoHash = "sha256-lTzEgy9mevkmefvTZT9hEBHN5I+kXVqTev5+sy/JoaE=";
+
+  meta = {
+    description = "Command-line tool to extract partitions from Android OTA files";
+    homepage = "https://github.com/crazystylus/otadump";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.axka ];
```

Comment:
missing `mainProgram`

Reply:
Hi, why is mainProgram needed when the pname is the same as the program's name? Is this a new policy in Nixpkgs?

Reply:
Yes, now all packages must have `mainProgram` set, if there is a main executable. See https://github.com/NixOS/nixpkgs/issues/281276 https://github.com/NixOS/nixpkgs/issues/219567